src/t_string.c

```diff
@@ -72,8 +72,8 @@ static int checkStringLength(client *c, long long size) {
 #define OBJ_PXAT (1<<7)            /* Set if timestamp in ms is given */
 #define OBJ_PERSIST (1<<8)         /* Set if we need to remove the ttl */
 
-void setGenericCommand(client *c, int flags, robj *key, robj *val, robj *expire, int unit, robj *ok_reply, robj *abort_reply) {
-    long long milliseconds = 0, when = 0; /* initialized to avoid any harmless warning */
+/* Forward declaration */
+static int getExpireMillisecondsOrReply(client *c, robj *expire, int flags, int unit, long long *milliseconds);
```

Comment:
there's something odd here.
i see that the commit that brought this change was a merge from unstable, but there was no such recent change in unstable, and further more, if you did merge from unstable, it should have probably brought many other changes.
i'm afraid that due to some error, the git log contains a merge, but the actual changes in the files got lost.
i'll try to look into it, but please try to think what you did wrong to cause it.

Reply:
This one i think it is ok now.
I take a look on my branch and the code. The git blame log is ok. nothing change
My merge step was:
- fetch the unstable branch
- in my branch, merge unstable
- then take care of the conflict (This conflict is not caused by my modification)

If there's a problem, i can reset back to the last commit. But there will still be conflicts.
The change came from #8474

Reply:
if you just `git fetch` and not `git pull`, you need to merge from `origin/unstable` rather than `unstable`.
but anyway, look at this commit in your branch: https://github.com/redis/redis/pull/8890/commits/f324ea9849015ac61980f7836115c8a307cba73c
it didn't bring any changes just some odd resolution for something that shouldn't have compiled.
and the actual changes from 8474 are not included.
maybe you forgot to `git add` something before committing.

the reason i worry is that it's possible to revert changes this way.
i.e. if you did a merge from unstable and in some way lost the actual changes in the files, the git log will think you hve these commits, and the changes will get lost.
if i just review your PR for updates (which i might because there's a lot of lines), and not re-review the full changes that are gonna be merged, i may miss that and we'll loose these.

Reply:
Sorry my mistake (Not clearly described), i did pull the unstable. And also i think i did `git add` before.
anyway, i will try to reset to `2a3e0864b73463b4239b3ce4133a74a045bfe7ce`. 
The one before `merge unstable conflicts`. And then squash the new commit(s) after it.
Then i will do a merge unstable(lastest one). 

It that ok with you? The new commits does not contains much changes
So i think we can try one more time to slove the conflict

Reply:
yeah, these are not a lot of changes, i can "review" them again.